lib/algo/DMDc.cpp

```diff
@@ -128,6 +128,7 @@ DMDc::DMDc(std::vector<std::complex<double>> eigs, Matrix* phi_real,
     d_k = k;
     d_dt = dt;
     d_t_offset = t_offset;
+    d_basis = basis;
```

Comment:
Since `d_basis` is getting set here in the constructor, we should probably check and free it when the class sets `d_basis` if the `train` method is called.

`train()` calls `createDMDc()` which sets `d_basis`. It would be good to add a check to free `d_basis` at https://github.com/LLNL/libROM/blob/124c6317afaa457f690264369f23caa10d523cdb/lib/algo/DMDc.cpp#L408 
```
if (d_basis)
{
    delete d_basis;
}
```